spatial/r3/vector_test.go

```diff
@@ -236,6 +236,34 @@ func TestCos(t *testing.T) {
 	}
 }
 
+func TestRotate(t *testing.T) {
+	const tol = 1e-14
+	for _, test := range []struct {
+		v, axis Vec
+		alpha   float64
+		want    Vec
+	}{
+		{Vec{1, 0, 0}, Vec{1, 0, 0}, math.Pi / 2, Vec{1, 0, 0}},
+		{Vec{1, 0, 0}, Vec{1, 0, 0}, 0, Vec{1, 0, 0}},
+		{Vec{1, 0, 0}, Vec{1, 0, 0}, 2 * math.Pi, Vec{1, 0, 0}},
+		{Vec{1, 0, 0}, Vec{0, 0, 0}, math.Pi / 2, Vec{math.NaN(), math.NaN(), math.NaN()}},
+		{Vec{1, 0, 0}, Vec{0, 1, 0}, math.Pi / 2, Vec{0, 0, -1}},
+		{Vec{1, 0, 0}, Vec{0, 1, 0}, math.Pi, Vec{-1, 0, 0}},
+		{Vec{2, 0, 0}, Vec{0, 1, 0}, math.Pi, Vec{-2, 0, 0}},
+		{Vec{1, 2, 3}, Vec{1, 1, 1}, 2. / 3. * math.Pi, Vec{3, 1, 2}},
+	} {
+		t.Run("", func(t *testing.T) {
+			got := test.v.Rotate(test.alpha, test.axis)
+			if !vecApproxEqual(got, test.want, tol) {
+				t.Fatalf(
```

Comment:
This should be an `Errorf`.

Reply:
It's in a subtest so it's ok to use Fatalf, isn't it?

Reply:
Does it need to me a subtest?

Reply:
consistency (with all the other `r3` and `r2` tests)?
I find it easier to run/test a particular test case with sub tests.

but if you prefer "flat" tests, I won't insist.

Reply:
I don't really see the point of them as subtests; they're not named and there are not groups of tests there. Deleting the `t.Run("", func(t *testing.T) {` and matching `})` leaves the syntax almost identical, but simpler.

Reply:
ok. do you want me to go in `r3` and "flatten" other `r3.Vec` tests as well? in this PR?

Reply:
I think in another PR.

Reply:
ok. done w/ PS6

Reply:
thanks